Release/include/cpprest/details/basic_types.h

```diff
@@ -43,20 +48,358 @@ typedef uint64_t size64_t;
 typedef uint32_t HRESULT; // Needed for PPLX
 #endif
 
+#if defined _WIN32 && !defined _DLL
+// TEMPLATE CLASS allocator
+template<class _Ty>
+class allocator : public std::allocator<_Ty> {
+public:
+    typedef allocator<_Ty> _Alloc;
+    typedef std::allocator<_Ty> _Mybase;
+
+    template<class _Other>
+    struct rebind {
+        // convert this type to allocator<_Other>
+        typedef allocator<_Other> other;
+    };
+
+    allocator() CPPREST_NOEXCEPT : _Mybase() {
+        // construct default allocator (do nothing)
+    }
+
+    allocator(const _Alloc& _Right) CPPREST_NOEXCEPT : _Mybase(_Right) {
+        // construct by copying (do nothing)
+    }
+
+    allocator(_Alloc&& _Right) CPPREST_NOEXCEPT : _Mybase(std::move(_Right)) {
```

Comment:
This constructor doesn't do anything.

Reply:
This is copy paste from `std::allocator`. I don't claim to know all `std::allocator` requirements, but if any function is not being used it will not be compiled right? In any case anything not being used can be removed.